libqpdf/QPDF_linearization.cc

```diff
@@ -121,13 +121,9 @@ QPDF::isLinearized()
         }
 
         QPDFTokenizer::Token t1 = readToken(this->m->file);
-        QPDFTokenizer::Token t2 = readToken(this->m->file);
```

Comment:
This actually changes the meaning -- the old code unconditionally reads 4 tokens, while the new code may or may not read tokens depending on whether earlier tokens pass their checks. That means that this change changes where the file points to after the function is called. Have you determined whether this is safe?

Reply:
There is a least one other example of that in this PR. Probably they are fine, but I haven't studied it.

Reply:
If the tokens don't pass the test either the loop is continued in which case another seek is called to explicitly set the file position before the next token is read, or the method returns false, in which case none of the callling code makes assumptions about the file position.